templates/terraform/post_create/dialogflow_agent_tier.go.erb

```diff
@@ -0,0 +1,25 @@
+
+// The tier field is eventually consistent, we need to test for consistency before moving on.
+// Otherwise, the user will see diffs on the field.
+if d.HasChange("tier") {
+	old, new := d.GetChange("tier")
+	readUrl, err := replaceVars(d, config, "{{DialogflowBasePath}}projects/{{project}}/agent")
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{old.(string)},
+		Target:  []string{new.(string)},
+		Refresh: func() (interface{}, string, error) {
+			res, err := sendRequest(config, "GET", project, readUrl, nil)
+			if err != nil {
+				return 0, "", err
+			}
+			return res, res["tier"].(string), nil
+		},
+		Timeout:                   40 * time.Minute,
```

Comment:
Oh - quick clarification. If the create/update timeout on the resource is less than this timeout it will kill the operation before this is reach.

Reply:
Unfortunately, my update test runs for 20+ minutes so I know it isn't being killed by the default timeout. Looking it up, those timeouts are only used for the actual create/update/delete [sendRequest](https://github.com/GoogleCloudPlatform/magic-modules/blob/c8abfb310171a90c1b933d0442bdd242b92de950/templates/terraform/resource.erb#L194) calls, which happens before my custom code is inserted

Reply:
At some point I remember seeing Terraform killing the whole sub process when the resource's timeout expired. Hopefully the timeout process is more nuanced than I thought and I'm not just mis-remembering. However the resource timeout is also an indication to the user of roughly how long an operation should take. Leaving those timeouts at 4 mins will show up in the documentation and then be confusing when the resource takes 20+ minutes. I think it's worth changing it either way.